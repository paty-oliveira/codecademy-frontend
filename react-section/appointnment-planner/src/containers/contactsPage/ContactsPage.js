import React, { useState, useEffect } from "react";

import { ContactForm } from '../../components/contactForm/ContactForm';
import { TileList } from '../../components/tileList/TileList';

export const ContactsPage = (props) => {

  const [name, setName] = useState('');
  const [phone, setPhone] = useState('');
  const [email, setEmail] = useState('');
  const [duplicates, setDuplicates] = useState(false);

  const contacts = props.contacts;
  const addNewContact = props.addContact;

  const handleSubmit = (e) => {
    e.preventDefault();

    if (!duplicates) {
      addNewContact(name, phone, email);
      setName('');
      setPhone('');
      setEmail('');
    }
  };

  useEffect(() => {
    const uniqueContactNames = new Set(contacts.map(contact => contact.name));
    let nameExists = false;

    if (uniqueContactNames.size < contacts.length){
      nameExists = true;
    }

    setDuplicates(nameExists);
  }, [contacts, name]);

  return (
    <div>
      <section>
        <h2>Add Contact</h2>
        <ContactForm
          name={name}
          phone={phone}
          email={email}
          setName={setName}
          setPhone={setPhone}
          setEmail={setEmail}
          handleSubmit={handleSubmit}
        />
      </section>
      <hr />
      <section>
        <h2>Contacts</h2>
        <TileList
          tile={contacts}
        />
      </section>
    </div>
  );
};
