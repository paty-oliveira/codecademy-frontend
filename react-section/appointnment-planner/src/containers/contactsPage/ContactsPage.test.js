import React from "react";
import { shallow } from 'enzyme';

import { ContactsPage } from "./ContactsPage";

describe('<ContactsPage />', () => {
    it('should render ContactsPage properly', () => {
        const contactsPageComponent = shallow(<ContactsPage />);

        expect(contactsPageComponent).toMatchSnapshot();
    });

    it('should render contacts with the given information', () => {
        const contact = [{
            name: "patricia",
            phoneNumber: "910025679",
            email: "example@xpto.com"
        }]

        const contactsPageComponent = shallow(<ContactsPage contacts={contact} />)

        expect(contactsPageComponent).toMatchSnapshot();
    });
});
