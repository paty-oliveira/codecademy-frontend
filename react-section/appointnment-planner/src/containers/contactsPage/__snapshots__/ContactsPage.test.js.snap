// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactsPage /> should render ContactsPage properly 1`] = `ShallowWrapper {}`;

exports[`<ContactsPage /> should render contacts with the given information 1`] = `ShallowWrapper {}`;
